sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -171,6 +174,17 @@ public EventData addProperty(String key, Object value) {
         return this;
     }
 
+    /**
+     * Creates a new {@link Context} object with all the keys and values provided
+     *
+     * @param context The key value pair object
+     * @return The updated EventData object.
+     */
+    public EventData context(Context context) {
+        this.context = context;
+        return this;
+    }
```

Comment:
@conniey Do you want to give your thoughts on this? Do you want Context passed in to the constructor or is this ok?

Reply:
I'd prefer the fields to be immutable, so having this as a constructor parameter is better, imho.

Reply:
In this, we have a scenario where we need the updated `EventData` object with the new span Context attached to it. But to do so, 

1. We would either need to add a method similar to `addProperty` to achieve it which returns the updated EventData object.
2. Or create a new `EventData` object from the parameterized constructor(with Context) using the existing event data object (overkill?)

Thoughts?